pubsub/rabbitmq/rabbitmq.go

```diff
@@ -190,10 +193,22 @@ func (r *rabbitMQ) publishSync(req *pubsub.PublishRequest) (rabbitMQChannelBroke
 		routingKey = val
 	}
 
+	ttl, ok, err := contribMetadata.TryGetTTL(req.Metadata)
+	if err != nil {
+		r.logger.Errorf("%s publishing to %s failed in TryGetTTL: %v", logMessagePrefix, req.Topic, err)
+		return r.channel, r.connectionCount, err
+	}
+	var expiration string
+	if ok {
+		// RabbitMQ expects the duration in ms
+		expiration = strconv.FormatInt(ttl.Milliseconds(), 10)
```

Comment:
Please check for negative TTL and a sane default must be set. 
Per documentation https://www.rabbitmq.com/ttl.html 
0 is expire immediately and negatives are not allowed ...

Reply:
`TryGetTTL` ensures the TTL is greater than 0(not included), and an empty `expiration` string will be ignored by the SDK when sent to the server.